cmd/run.go

```diff
@@ -185,35 +185,27 @@ func (c *cmdRun) run(cmd *cobra.Command, args []string) error {
 
 	// Start the test run
 	initBar.Modify(pb.WithConstProgress(0, "Starting test..."))
-	var interrupt error
 	err = engineRun()
 	if err != nil {
-		err = common.UnwrapGojaInterruptedError(err)
-		if errext.IsInterruptError(err) {
-			// Don't return here since we need to work with --linger,
-			// show the end-of-test summary and exit cleanly.
-			interrupt = err
-		}
-		if !conf.Linger.Bool && interrupt == nil {
-			return errext.WithExitCodeIfNone(err, exitcodes.GenericEngine)
-		}
+		logger.WithError(err).Debug("Engine terminated with an error")
+	} else {
+		logger.Debug("Engine run terminated cleanly")
 	}
 	runCancel()
-	logger.Debug("Engine run terminated cleanly")
 
 	progressCancel()
 	progressBarWG.Wait()
 
 	executionState := execScheduler.GetState()
 	// Warn if no iterations could be completed.
-	if executionState.GetFullIterationCount() == 0 {
+	if err == nil && executionState.GetFullIterationCount() == 0 {
 		logger.Warn("No script iterations finished, consider making the test duration longer")
 	}
 
 	// Handle the end-of-test summary.
-	if !testRunState.RuntimeOptions.NoSummary.Bool {
+	if err == nil && !testRunState.RuntimeOptions.NoSummary.Bool {
```

Comment:
Do we not want to output the summary even if `test.abort()` was called? We currently do on `master` and it would be useful to output whatever data we got before the test was aborted.

Reply:
Great catch! This is a regression from this PR, I'll fix it and add it to the test.

That said, I'd say that the current `master` code also has a bug. With it, this example also won't produce a summary, which is probably not what we want either:
```js
import http from 'k6/http';

export default function () {
    http.get('https://test.k6.io');
};

export function teardown() {
    console.log('wonky teardown');
    //throw new Error('foo');
}
``` 

So, I'll adjust the code so that k6 tries to print the `handleSummary()` even if there was an error during the test run. The only time `handleSummary()` won't be called is when it was explicitly disabled with `--no-summary`, or when the init code wasn't successful (because that probably means there is a script error in the init context and k6 won't be able to initialize a transient VU to run `handleSummary()` anyway)

WDYT?

Reply:
A bit bad that no test failed for this. Could we move the logic as a method of `testRunState` so we could centralize it and cover it with a unit test?